### PR TITLE
Use re.fullmatch when available

### PR DESCRIPTION
**Pull request by Tasssadar**
Python's `re` has `fullmatch` method built-in since 3.4, and it is much faster than re-compiling the patterns all the time. The speed-up is visible even when running tests, they are 1s faster with this.

Also removing the `flags` parameter, which was unused and broken.

### CODE REVIEW

File: python/phonenumbers/re_util.py [hunk 1]
```diff
@@ -22,23 +22,35 @@
...
+        grouped_pattern = re.compile("^(?:%s)$" % pattern.pattern, pattern.flags)
+        m = grouped_pattern.match(string)
+        if m and m.end() < len(string):
+            # Incomplete match (which should never happen because of the $ at the
+            # end of the regexp), treat as failure.
+            m = None  # pragma no cover
+        return m
+
+
+if __name__ == "__main__":  # pragma no cover
```
Comment (daviddrysdale) [inline python/phonenumbers/re_util.py]: nit: is there a reason to change quote style and add blank lines?

**Review by daviddrysdale — APPROVED** [review]
Thanks!

File: python/phonenumbers/re_util.py (hunk 1 above)
Comment (Tasssadar) [reply python/phonenumbers/re_util.py]: nope, autoformatting - fixed now, sorry.

**Review by daviddrysdale — APPROVED** [review]

**Comment by daviddrysdale** [comment]
Thanks!

**Pull request merged**